csharp/source/autowrap/csharp/common.d

```diff
@@ -77,5 +66,110 @@ package string getDLangSliceInterfaceName(string fqn, string funcName) {
 public string camelToPascalCase(string camel) {
     import std.uni : toUpper;
     import std.conv : to;
-    return to!string(camel[0].toUpper) ~ camel[1..$];
+    import std.range : dropOne;
+    return to!string(camel.front.toUpper()) ~ camel.dropOne();
+}
+
+package template verifySupported(T)
+{
+    static if(isSupportedType!T)
+        enum verifySupported = true;
+    else
+    {
+        pragma(msg, T.stringof ~ " is not currently supported by autowrap's C#");
+        enum verifySupported = false;
+    }
+}
+
+package template numDefaultArgs(alias func)
+{
+    import std.meta : Filter;
+    import std.traits : ParameterDefaults;
+
+    enum numDefaultArgs = Filter!(isNotVoid, ParameterDefaults!func).length;
+
+    private static template isNotVoid(values...)
+        if(values.length == 1)
+    {
+        enum isNotVoid = !is(values[0] == void);
+    }
+}
+
+// Unfortunately, while these tests have been tested on their own, they don't
+// currently run as part of autowrap's tests, because dub test doesn't work for
```

Comment:
That's because dub is a bit... let's say "not ideal" when it comes to subpackages. See the unittest configuration for the top-level dub.sdl. It's annoying, but it's the only way I know how to run the unit tests.